src/op_mode/nhrp.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2023 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import tabulate
+import vyos.opmode
+
+from vyos.util import cmd
+from vyos.util import process_named_running
+
+
+def get_formatted_output(output_list: list) -> str:
+    """
+    Create formatted table for CLI output
+    :param output_list: list for API
+    :type output_list: list
+    :return: tabulate string
+    :rtype: str
+    """
+    status: dict = output_list.pop(0)
+    if 'Status' in status:
+        print(f"Status: {status['Status']}")
+    output: str = tabulate.tabulate(output_list, headers='keys',
+                                    numalign="left")
+    return output
+
+
+def get_formatted_list(output_string: str) -> list:
+    """
+    Format string returned from CMD to API list
+    :param output_string: String received by CMD
+    :type output_string: str
+    :return: list for API
+    :rtype: list
+    """
+    formatted_list: list = []
+    output_list: list = output_string.split('\n\n')
+    for list_a in output_list:
+        lists: list = list_a.split('\n')
+        output_dict: dict = {}
+        for entry in lists:
+            delimiter_index: int = entry.find(':')
+            if delimiter_index != -1 and delimiter_index != len(entry) - 1:
+                key: str = entry[:delimiter_index].strip()
+                value: str = entry[delimiter_index + 1:].strip()
+                output_dict[key] = value
+        formatted_list.append(output_dict)
+    return formatted_list
+
+
+def show_interface(raw: bool):
+    """
+    Command 'show nhrp interface'
+    :param raw: if API
+    :type raw: bool
+    """
+    if not process_named_running('opennhrp'):
+        raise vyos.opmode.UnsupportedOperation('OpenNHRP is not running.')
```

Comment:
We have a dedicated exception specially for this situation: [`UnconfiguredSubsystem`](https://github.com/vyos/vyos-1x/blob/current/python/vyos/opmode.py#L32-L36). `UnsupportedOperation` is for operations that the current VyOS version doesn't support at all, e.g., when a tool lacks machine-readable output for some of the commands.

Reply:
done